pkg/alicloud/client/client.go

```diff
@@ -198,6 +198,24 @@ func (c *ecsClient) CheckIfImageExists(ctx context.Context, imageID string) (boo
 	return response.TotalCount > 0, nil
 }
 
+// CheckIfImagePublic checks whether given imageID can is public image
+func (c *ecsClient) CheckIfImagePublic(ctx context.Context, imageID string) (bool, error) {
```

Comment:
Please enhance CheckIfImageExists method directly. Please check it also works for custom shared image